packages/react-native/ReactCommon/react/nativemodule/samples/platform/android/SampleTurboModule.java

```diff
@@ -245,4 +246,12 @@ public void invalidate() {}
   public String getName() {
     return NAME;
   }
+
+  @Override
+  @DoNotStrip
+  public void installJSIBindings(long runtime) {
+    installJSIBindingsCxx(runtime);
+  }
```

Comment:
so it feels like one motivation of having this live in native land is that we might have some native dependencies that we want to pass down the the runtime bindings. if that's the case, i feel like this will be called before we could do anything to manipulate the native state of the module, or do you see it differently?

Reply:
> if that's the case, i feel like this will be called before we could do anything to manipulate the native state of the module

from js side, yes, and that's why we need a synchronous way to install bindings right after initiating the module and before the further js call. i.e.

```js
const module = TurboModuleRegistry.getEnforcing<Spec>('Module');
// globalThis.someJSIBindings should be available here
module.call();
```